dspace-api/src/main/java/org/dspace/content/authority/ChoiceAuthorityServiceImpl.java

```diff
@@ -249,68 +301,89 @@ private void loadChoiceAuthorityConfigurations() {
+                    DCInput[][] dcinputs = dcinputSet.getFields();
+                    for (DCInput[] dcrows : dcinputs) {
+                        for (DCInput dcinput : dcrows) {
+                            if (StringUtils.isNotBlank(dcinput.getPairsType())
+                                || StringUtils.isNotBlank(dcinput.getVocabulary())) {
+                                String authorityName = dcinput.getPairsType();
+                                if (StringUtils.isBlank(authorityName)) {
+                                    authorityName = dcinput.getVocabulary();
+                                }
+                                if (!StringUtils.equals(dcinput.getInputType(), "qualdrop_value")) {
+                                    String fieldKey = makeFieldKey(dcinput.getSchema(), dcinput.getElement(),
+                                                                   dcinput.getQualifier());
+                                    ChoiceAuthority ca = controller.get(authorityName);
+                                    if (ca == null) {
+                                        ca = (ChoiceAuthority) pluginService
+                                            .getNamedPlugin(ChoiceAuthority.class, authorityName);
+                                        if (ca == null) {
+                                            throw new IllegalStateException("Invalid configuration for " + fieldKey
+                                                    + " in submission definition " + submissionName
+                                                    + ", form definition " + dcinputSet.getFormName()
+                                                    + " no named plugin found: " + authorityName);
+                                        }
                                     }
 
-                                    ChoiceAuthority ma = (ChoiceAuthority) pluginService
-                                        .getNamedPlugin(ChoiceAuthority.class, authorityName);
-                                    if (ma == null) {
-                                        log.warn("Skipping invalid configuration for " + fieldKey
-                                                     + " because named plugin not found: " + authorityName);
-                                        continue;
+                                    Map<String, ChoiceAuthority> definition2authority;
+                                    if (controllerFormDefinitions.containsKey(fieldKey)) {
+                                        definition2authority = controllerFormDefinitions.get(fieldKey);
+                                    } else {
+                                        definition2authority = new HashMap<String, ChoiceAuthority>();
+                                    }
+                                    definition2authority.put(submissionName, ca);
+                                    controllerFormDefinitions.put(fieldKey, definition2authority);
```

Comment:
hi @tdonohue 
no this is used to fix the previous implementation where essentially we have forced all the metadata that were linked to a value-pairs to became really authority controlled (with also the side effect to store the authority) regardless to the collection. With this more flexible structure we know can discriminate between real authority controlled metadata and metadata that for specific collection (or better collection definitions) require controlled vocabularies (without storing the authority).

The collection and metadata params are required for the existing method signatures in the ChoiceAuthority that has been retained to avoid additional refactoring (see also my comment in the rest contract https://github.com/DSpace/Rest7Contract/pull/128#discussion_r446911316 )